libs/fills/src/lib/fills-manager.tsx

```diff
@@ -22,26 +23,68 @@ export const FillsManager = ({
   const filter: Schema.TradesFilter | Schema.TradesSubscriptionFilter = {
     partyIds: [partyId],
   };
-  const { data, error } = useDataProvider({
+  const [hasFilteredRows, setHasFilteredRowsCount] = useState<
+    boolean | undefined
+  >(undefined);
+  const { data, error, load, pageInfo } = useDataProvider({
     dataProvider: fillsWithMarketProvider,
-    update: ({ data }) => {
-      if (data?.length && gridRef.current?.api) {
-        gridRef.current?.api.setRowData(data);
-        return true;
-      }
-      return false;
-    },
     variables: { filter },
   });
-
+  const { onFilterChanged, ...props } = gridProps;
+  const onRowDataUpdated = useCallback(
+    ({ api }: { api: AgGridReact['api'] }) => {
+      let hasFilteredRows = false;
+      try {
+        api.forEachNodeAfterFilter(() => {
+          throw new Error();
+        });
+      } catch (e) {
+        hasFilteredRows = true;
+      }
+      setHasFilteredRowsCount(hasFilteredRows);
+    },
+    []
+  );
   return (
-    <FillsTable
-      ref={gridRef}
-      rowData={data}
-      partyId={partyId}
-      onMarketClick={onMarketClick}
-      overlayNoRowsTemplate={error ? error.message : t('No fills')}
-      {...gridProps}
-    />
+    <div className="flex flex-col h-full">
+      <FillsTable
+        ref={gridRef}
+        rowData={data}
+        onFilterChanged={(event) => {
+          onRowDataUpdated(event);
+          onFilterChanged(event);
+        }}
+        onRowDataUpdated={onRowDataUpdated}
+        partyId={partyId}
+        onMarketClick={onMarketClick}
+        overlayNoRowsTemplate={error ? error.message : t('No fills')}
+        {...props}
+      />
+
+      <div className="flex justify-between border-t border-default p-1 items-center">
+        <div className="text-xs">
+          {t(
+            'Depending on data node retention you may not be able see the "full" history'
+          )}
+        </div>
+        <div className="flex text-xs items-center">
+          {data?.length && !pageInfo?.hasNextPage
+            ? t('all %s items loaded', [data.length.toString()])
+            : t('%s items loaded', [
+                data?.length ? data.length.toString() : ' ',
+              ])}
+          {pageInfo?.hasNextPage ? (
+            <Button size="extra-small" className="ml-1" onClick={() => load()}>
+              {t('Load more')}
+            </Button>
+          ) : null}
+        </div>
+        {data?.length && hasFilteredRows === false ? (
```

Comment:
Can we not use overlayNoRowsTemplate? Is it rendered if there are rows in memory but none after filtering?

Reply:
Ag-grid is not showing no rows overlay if there are no rows to display because of filters applied.